nkms/crypto/powers.py

```diff
@@ -94,6 +94,59 @@ def encrypt_for(self, pubkey, cleartext):
         except KeyError:
             raise NoEncryptingPower
 
+    def combine(self, cfrags):
+        try:
+            encrypting_power = self._power_ups[EncryptingPower]
+        except KeyError:
+            raise NoEncryptingPower
+        the_secret = encrypting_power.combine(cfrags)
+        return the_secret
+
+    def decrypt_reencrypted(
+            self,
+            edata: Tuple[Tuple[bytes, bytes], Tuple[bytes, bytes]]
+            ) -> bytes:
+        """
+        Decrypt data which was re-encrypted for our public key
+        Format of edata is the same as the output of reencrypt() method:
+        data encrypted with an ephemeral key
+        and the ephemeral private key encrypted for recepient (Bob)
+        """
+        edata_by_eph, encrypted_eph = edata
+        priv_eph = self._decrypt(encrypted_eph)
+        return self._decrypt(edata_by_eph, privkey=priv_eph)
+
+    def _decrypt(self,
```

Comment:
Btw this doesn't have to be underscored. At the end of the day, Alice uses this to read her own data

Reply:
Also am I right that this will go to `keypair`?